server/src/Service/InteractionService.php

```diff
@@ -142,4 +145,18 @@ private function sanitizeDate( $date ) {
 
 		return isset( $dateObj ) ? $dateObj : null;
 	}
+
+	/**
+	 * Check if a value in an array is a valid IP address
+	 *
+	 * @param array $data
+	 * @return boolean
+	 */
+	private function isIpPresent( $data ) {
+		$ips = array_filter( $data, function ( $possibleIp ) {
+			return filter_var( $possibleIp, FILTER_VALIDATE_IP );
+		} );
+
+		return count( $ips ) > 0;
```

Comment:
nit-pick: I think you could just do something like:
```php
return (boolean) array_filter( $data, function ( $possibleIp ) {
  return filter_var( $possibleIp, FILTER_VALIDATE_IP );
} );
```
but if that's harder to read, then what you have is fine.

Reply:
I think count is clearer tho. `array_filter` returns an array and casting an array to boolean feels weird to me.